src/Security/AuthenticationRequestFilter.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace SilverStripe\Security;
+
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\Control\RequestFilter;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\Session;
+use SilverStripe\ORM\DataModel;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\ORM\ValidationException;
+
+class AuthenticationRequestFilter implements RequestFilter, IdentityStore
+{
```

Comment:
@tractorcow
Although I agree, I don't think it's current form would allow for that. I would happily discuss it with you and see where we could get to. I don't see much possibility on that perspective here personally.